metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/StringFunctions.java

```diff
@@ -284,4 +287,36 @@ private static Object fill(FillDirection direction, Object inputObject, Object f
     }
     return org.apache.commons.lang.StringUtils.rightPad(input,requiredLength,fill);
   }
+
+  @Stellar( namespace="STRING"
+          , name="ENTROPY"
+          , description = "Computes the base-2 shannon entropy of a string"
```

Comment:
Good catch; this is, at this point, pretty classical work, so I link to the wikipedia definition rather than claude shannon's original work.